management_api_app/validation/validate_request.py

```diff
@@ -0,0 +1,40 @@
+from fastapi import HTTPException
+from starlette import status
+
+from db.errors import EntityDoesNotExist
+from db.repositories.workspaces import WorkspaceTemplateRepository
+from models.schemas.workspace import WorkspaceInCreate
+from resources import strings
+
+
+class ValidateRequest:
+    """
+    Implements methods to validate API requests.
+    """
+
+    @staticmethod
+    def validate_request(workspace_create: WorkspaceInCreate, workspace_template_repo: WorkspaceTemplateRepository):
+        validation_errors = []
+        try:
+            template = workspace_template_repo.get_current_workspace_template_by_name(workspace_create.workspaceType)
+        except EntityDoesNotExist:
+            validation_errors.append(strings.WORKSPACE_TEMPLATE_X_DOES_NOT_EXIST.format(workspace_create.workspaceType))
+
+        if not validation_errors:
+            template_parameters = template.get(strings.VALIDATION_PARAMETERS)
+            validation_errors = ValidateRequest.validate_parameters(template_parameters, workspace_create.parameters)
+
+        if validation_errors:
+            raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST,
+                                detail={strings.VALIDATION_ERRORS: validation_errors})
+
+    @staticmethod
+    def validate_parameters(template_parameters: [dict], workspace_create_parameters: dict):
+        validation_errors = []
+        for parameter in template_parameters:
+            name = parameter.get(strings.VALIDATION_NAME)
+            create_parameter = workspace_create_parameters.get(name)
+            if create_parameter is None and parameter.get(strings.VALIDATION_REQUIRED):
+                validation_errors.append(strings.PARAMETER_X_IS_REQUIRED.format(name))
```

Comment:
Please rewrite and validate that this works (I am just writing a suggestion from memory here)
```suggestion
        for parameter in template_parameters:
            name = parameter.name
            create_parameter = workspace_create_parameters[name]
            if parameter.required and not create_parameter:
                validation_errors.append(strings.PARAMETER_X_IS_REQUIRED.format(name))
```